src/Symfony/Component/Form/Tests/Extension/Core/Type/CollectionTypeTest.php

```diff
@@ -177,6 +177,29 @@ public function testResizedDownIfSubmittedWithCompoundEmptyDataAndDeleteEmpty()
         $this->assertEquals(array(new Author('s_first', 's_last')), $form->getData());
     }
 
+    public function testResizedDownIfSubmittedWithCompoundEmptyDataDeleteEmptyAndNoDataClass()
+    {
+        $form = $this->factory->create(static::TESTED_TYPE, null, array(
+            'entry_type' => 'Symfony\Component\Form\Tests\Fixtures\AuthorType',
+            // If the field is not required, no new Author will be created if the
+            // form is completely empty
+            'entry_options' => array('data_class' => null),
+            'allow_add' => true,
+            'delete_empty' => true,
```

Comment:
Could you add `'allow_delete' => true` and try again? (https://github.com/symfony/symfony/issues/22008#issuecomment-286836414)